### PR TITLE
ocrvs-2290 Fixed the types used in ApplicationForm and RegisterForm

### PR DESCRIPTION
**Pull request by Zangetsu101**

### CODE REVIEW

**Review by euanmillar — APPROVED** [review]
Looks fine to me.  @yeasinhossain-dsi and @tahmidrahman-dsi can you take a look please?  

@Zangetsu101 you said it was buggy. Can you explain the bug?

File: packages/client/src/views/RegisterForm/ApplicationForm.tsx [hunk 1]
```diff
@@ -70,14 +63,6 @@ function mapStatetoProps(
   }
 }
 
-export const ApplicationForm = connect<
-  StateProps,
-  {},
-  IFormProps &
```
Comment (Zangetsu101) [inline packages/client/src/views/RegisterForm/ApplicationForm.tsx:76]: The `ownProps` for `ApplicationForm` is only `RouteComponentProps` as this Component is rendered directly from `<Route />` so IFormProps should not be here.

File: packages/client/src/views/RegisterForm/ApplicationForm.tsx [hunk 2]
```diff
@@ -23,29 +23,22 @@ import {
...
+
+export class ApplicationFormView extends React.Component<
+  IFormProps & RouteProps
+> {
   render() {
     return <RegisterForm {...this.props} />
   }
 }
 
-interface StateProps {
```
Comment (Zangetsu101) [inline packages/client/src/views/RegisterForm/ApplicationForm.tsx:39]: `StateProps` is actually similar to `IFormProps` so it's redundant.

File: packages/client/src/views/RegisterForm/ApplicationForm.tsx [hunk 3]
```diff
@@ -23,29 +23,22 @@ import {
...
 import { connect } from 'react-redux'
-import { Event, IForm } from '@client/forms'
-import { IApplication } from '@client/applications'
+import { Event } from '@client/forms'
 
 const pageRoute: { [key in Event]: string } = {
   birth: DRAFT_BIRTH_PARENT_FORM_PAGE_GROUP,
   death: DRAFT_DEATH_FORM_PAGE_GROUP
 }
-export class ApplicationFormView extends React.Component<FullProps> {
```
Comment (Zangetsu101) [inline packages/client/src/views/RegisterForm/ApplicationForm.tsx:33]: The props for this component should only be the ones returned from `mapStateToProps`  that is `IFormProps` and the `RouteComponentProps`, not `FullProps` .

Comment (euanmillar) [reply packages/client/src/views/RegisterForm/ApplicationForm.tsx:33]: Are

```
application: IApplication
  registerForm: IForm
```

not used at all in this component or its descendants?

Comment (euanmillar) [reply packages/client/src/views/RegisterForm/ApplicationForm.tsx:33]: My concern is that they might be used in RegisterForm and just passed through

Comment (Zangetsu101) [reply packages/client/src/views/RegisterForm/ApplicationForm.tsx:33]: Yes, they are used in RegisterForm and they are being passed through from ApplicationForm without any problem.

**Review by tahmidrahman-dsi — APPROVED** [review]
LGTM

**Pull request merged**